### PR TITLE
prepare dataset release & docs updates

### PR DESCRIPTION
**Pull request by sh-rp**
### Description
Makes dataset interface public and updates transformation and dataaccess snippets in docs to use this interface.

Note: For linting to pass on this PR, ibis dataset accessing must be merged first.

Stuff that happened:
* Remove sql_client reference in docs where not needed
* Updated sql_client sql queries to something more useful where it makes sense and added info about accessing data with datasets
* Reworked the transformations section
* Currently not yet added any datasets stuff to our quick starts, only to the rest_api tutorial. @rudolfix should I add this there?

### CODE REVIEW

**Review by rudolfix — APPROVED** [review]
OK, this is very good

**Review by rudolfix — CHANGES_REQUESTED** [review]
OK let's merge ibis first to see if docs will build again.

I'd show how to get data frame in https://dlthub.com/docs/intro "Python data structures" tab - to read content of the table as df

**Review by rudolfix — COMMENTED** [review]
LGTM. but let's wait for the final feedback

**Review by AstrakhantsevaAA — COMMENTED** [review]
Looks good! Suggest some small corrections

File: docs/website/docs/intro.md [hunk 1]
```diff
@@ -139,6 +151,10 @@ pipeline = dlt.pipeline(
 )
 
 load_info = pipeline.run(foo)
+
+# print load info and the "foo_data" table as dataframe
+print(load_info)
+print(pipeline.dataset().foo_table.df())
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/intro.md:157]:
```suggestion
print(pipeline.dataset().foo_data.df())
```

**Review by AstrakhantsevaAA — COMMENTED** [review]
noticed a typo

**Pull request merged**